packages/devextreme/js/__internal/ui/splitter/utils/splitter.layout_helper.ts

```diff
@@ -0,0 +1,122 @@
+import $ from '@js/core/renderer';
+import {
+  normalizeStyleProp, styleProp,
+} from '@js/core/utils/style';
+
+const ORIENTATION = {
+  horizontal: 'horizontal',
+  vertical: 'vertical',
+};
+
+const FLEX_PROPERTY_NAME = 'flexGrow';
+const RESIZE_HANDLE_CLASS = 'dx-resize-handle';
+const DEFAULT_RESIZE_HANDLE_SIZE = 8;
+
+export default class SplitterLayoutHelper {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private readonly items: any;
+
+  private readonly orientation: string;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private readonly $element: any;
+
+  private layoutState: number[];
+
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  constructor(items: unknown, orientation: string, $element) {
+    this.items = items;
+    this.$element = $element;
+    this.layoutState = [];
+    this.orientation = orientation;
+  }
+
+  initializeState(): void {
+    this.layoutState = [];
+
+    this.items.each((index, splitterItem) => {
+      // @ts-expect-error todo: fix error
+      this.layoutState.push(parseFloat($(splitterItem).css(FLEX_PROPERTY_NAME)));
+      //
+    });
+  }
+
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  applyNewLayout(e): void {
+    const newLayoutState = this._getNewLayout(e);
+
+    this.items.each((index, item) => {
+      this._setFlexProp(item, FLEX_PROPERTY_NAME, newLayoutState[index]);
+    });
+  }
+
+  layoutItems(): void {
+    // NOTE: temporarily evenly distributed
+    const splitterItemRatio = 100 / this.items.length;
+
+    this.items.each((index, item) => {
+      this._setFlexProp(item, FLEX_PROPERTY_NAME, splitterItemRatio);
+    });
+  }
+
+  // eslint-disable-next-line max-len
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, class-methods-use-this
+  _setFlexProp(element, prop, value): void {
+    const normalizedProp = normalizeStyleProp(prop, value);
+    element.style[styleProp(prop)] = normalizedProp;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  _getOffset(e): number {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return this.orientation === ORIENTATION.horizontal ? e.offset.x : e.offset.y;
+  }
+
+  _getSplitterItemsSizeSum(): number {
+    const splitterSize = this.$element.get(0).getBoundingClientRect();
```

Comment:
If we use flex-grow for panes sizes and flex-bases for handles we shouldn't think about resize handle width

Reply:
i tested it and seems like it is required anyway